pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -322,14 +403,33 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 			return nil
 		} else /* pv != nil */ {
 			// Found a PV for this claim
+			// There are three scenarios here:
+			// 1. the PV is newly dynamically provisioned
+			// 2. an existing PV meets the requirement of the PVC and available
+			// 3. user set "claimRef" of a PV to the PVC
+			// for metric reporting, the first case will be provision operation latency, and the
+			// other two would be binding, if there exists NO provision timestamp AND the volume
+			// got NO annotation of pvutil.annDynamically, its a binding operation, insert
+			// a timestamp for start bindings
+			claimKey := claimToClaimKey(claim)
+			metricOperation := "provision"
+			if !ctrl.existsOperationTimestamp("provision", claimKey) && !metav1.HasAnnotation(volume.ObjectMeta, pvutil.AnnDynamicallyProvisioned) {
+				ctrl.addOperationTimestamp("binding", claimKey, ctrl.getProvisionerNameFromVolume(volume))
```

Comment:
If there already is a provision timestamp, when do we delete it from the cache?

Reply:
Can we start the clock before `findBestMatchForClaim`? This function has to iterate over all PVs in the system, so can contribute to latency.

Reply:
> If there already is a provision timestamp, when do we delete it from the cache?
when a volume has been provisioned successfully and bound to the claim, or the claim has been deleted from etcd.



> Can we start the clock before `findBestMatchForClaim`? This function has to iterate over all PVs in the system, so can contribute to latency.

will accommodate

Reply:
Ah sorry... I realized putting it before `findBestMatchForClaim` may make things more complicated.

Because it's possible we won't find any match due to user configuration error, ie they don't have the PVs created or it's delayed binding and they haven't created a pod yet.

So for simplicity, I think ti's probably better to leave the start time where it was

Reply:
it does introduce a bit confusion and will make the code messy. also the timestamp will only be used when a pv has actually been found, AND the provisioning operation does not exist

Reply:
no longer apply